c10/core/DispatchKeySet.cpp

```diff
@@ -46,14 +46,13 @@ bool isBackendDispatchKey(DispatchKey t) {
 // math_dispatch_keyset contains all keys in backend_dispatch_keyset and
 // autograd_dispatch_keyset Alias key DispatchKey::CompositeImplicitAutograd
 // maps to [math_dispatch_keyset x full_backend_mask]
-constexpr DispatchKeySet math_dispatch_keyset = backend_dispatch_keyset |
-    autograd_dispatch_keyset |
-    // See Note [NestedTensor Not Included in Backend Keys]
-    // The caveat to that note is that nested_tensor is a special case
-    // where we would like to support composite implicit kernels but not
-    // explicit kernels therefore we manually add the key to the
-    // math_dispatch_keyset
-    DispatchKeySet{DispatchKey::NestedTensor};
+constexpr DispatchKeySet math_dispatch_keyset =
```

Comment:
Did this remove the ability for nested tensors to run in "regular" Composite kernels. We should have both right? And just have CompositeImplicitAutogradNestedTensor take precedence. @bdhirsh

Reply:
I think yes, but I have modified codegen in such a way that CompositeImplicitAutogradNetsedTensor kernel got registered even if only CompositeImplicitAutograd is specified. Take `softmax` as an example, in native_functions.yaml it has:
```
- func: softmax.int(Tensor self, int dim, ScalarType? dtype=None) -> Tensor
  variants: function, method
```
which makes it CompositeImplicitAutograd. A `torch._C._dispatch_dump("aten::softmax.int")` call gives us:
```
name: aten::softmax.int
schema: aten::softmax.int(Tensor self, int dim, int? dtype=None) -> Tensor
debug: registered at aten/src/ATen/RegisterSchema.cpp:6
alias analysis kind: FROM_SCHEMA
Functionalize: registered at aten/src/ATen/RegisterFunctionalization_2.cpp:20485 :: (Tensor _0, int _1, int? _2) -> Tensor _0 [ boxed unboxed ]
Named: registered at ../aten/src/ATen/core/NamedRegistrations.cpp:11 :: (none) [ fallthrough boxed ]
Tracer: registered at ../torch/csrc/autograd/generated/TraceType_3.cpp:14597 :: (Tensor _0, int _1, int? _2) -> Tensor _0 [ boxed unboxed ]
AutocastCUDA: registered at ../aten/src/ATen/autocast_mode.cpp:328 :: (Tensor _0, int _1, int? _2) -> Tensor _0 [ boxed unboxed ]
CompositeImplicitAutograd[alias]: registered at aten/src/ATen/RegisterCompositeImplicitAutograd.cpp:6983 :: (Tensor _0, int _1, int? _2) -> Tensor _0 [ boxed unboxed ]
CompositeImplicitAutogradNestedTensor[alias]: registered at aten/src/ATen/RegisterCompositeImplicitAutogradNestedTensor.cpp:6922 :: (Tensor _0, int _1, int? _2) -> Tensor _0 [ boxed unboxed ]
```
Which has CompositeImplicitAutogradNestedTensor registered as well

Reply:
Got it so the registered kernel is the same for CompositeImplicitAutograd , CompositeImplicitAutogradNestedTensor when CompositeImplicitAutogradNestedTensor is not specified? This makes sense

Reply:
hmm I'm not sure that this is what we want.

I think of `CompositeImplicitAutograd` as "all backends + layout keys, plus all autograd keys".

And then now, I think of `CompositeImplicitAutogradNestedTensor` as "all nested backends, plus the nested autograd key".

So in that view, `CompositeImplicitAutograd` should be a strict superset of `CompositeImplicitAutogradNestedTensor`: if a kernel is registered directly to `CompositeImplicitAutograd`, that should still cause all of the nested tensor dispatch keys to get that kernel too.

We also have a handful of native_function entries that explicit use `CompositeImplicitAutograd`, so this would cause NestedTensor to lose those kernels. e.g. https://github.com/pytorch/pytorch/blob/1f38225b56b873c944196241ea448445a61798fd/aten/src/ATen/native/native_functions.yaml#L3951.

Reply:
Have added NestedTensor back to math keyset